### PR TITLE
Import point-cloud metadata - schema and CRS

### PR DESCRIPTION
**Pull request by olsen232**
- import meta/schema.json and meta/crs/<CRS_ID>.wkt during point cloud
  import.
- also:
  = leave COPC as COPC
  = optionally convert LAZ to COPC (default is yes)
  = always convert LAS to COPC

https://github.com/koordinates/kart/issues/565

### CODE REVIEW

File: kart/point_cloud/import_.py [hunk 1]
```diff
@@ -34,11 +34,12 @@
 
 @click.command("point-cloud-import", hidden=True)
 @click.pass_context
+@click.option("--convert-to-copc/--no-convert-to-copc", is_flag=True, default=True)
```
Comment (craigds) [inline kart/point_cloud/import_.py:37]: not sure why this is an option? I would have thought we'd always want everything in COPC. Why would a user [be able to] disable this?

If it's just for some testing use-case, should add hidden=True to avoid 'supporting' this.

If it's for performance reasons (don't want import to be slow) we might be shooting ourselves in the foot since it'll presumably make data consumption slower later?

File: kart/point_cloud/import_.py [hunk 2]
```diff
@@ -124,8 +137,39 @@ def point_cloud_import(ctx, ds_path, sources):
             "crs84_envelope": crs84_envelope,
         }
 
+        if schema is None:
+            crs_name = get_identifier_str(crs_set[0])
```
Comment (craigds) [inline kart/point_cloud/import_.py:141]: minor thing but: these things are ListBasedSet instances - I understand the reason is so you can have a set that stores non-hashable things; ok fair enough

but then you're accessing them here quite unlike a set, relying on the underlying type as well as the ordering. I'd prefer `list(crs_set)[0]` or something I guess. Maybe you shouldn't be able to index a ListBasedSet at all since it's trying to be like a set

File: kart/point_cloud/import_.py [hunk 3]
```diff
@@ -124,8 +137,39 @@ def point_cloud_import(ctx, ds_path, sources):
...
+            schema = metadata["filters.info"]["schema"]
+            schema["CRS"] = crs_name
+
     click.echo()
 
+    version = version_set[0]
```
Comment (craigds) [inline kart/point_cloud/import_.py:147]: and, for these ones where they _don't_ store dicts, maybe just use a normal `set()` ?

File: kart/point_cloud/import_.py (hunk 1 above)
Comment (olsen232) [reply kart/point_cloud/import_.py:37]: This spec has been provided by Rob himself. The original plan was to only allow COPC since it's the cooler format that lets us do more stuff - but, newer plan (IIUC) is that sometimes you'll be able to turn a pile of LAZ files that you have in eg S3 into a Kart repo somehow.  I haven't got to that bit of the plan yet - focussing on local only still - but as a concession to it, I'm already allowing repos backed with non-COPC LAZ files (which are commonly used in the real world, but not LAS files, which mostly aren't).

File: kart/point_cloud/import_.py (hunk 2 above)
Comment (olsen232) [reply kart/point_cloud/import_.py:141]: I made a function .only() that does the same as [0] except that it asserts that it's a single-element-set, and I removed the __getitem__ operator.
It's a well-known fact that the first element of a single-element-set is reliably the only element, and that you can get it with next(iter(myset)) but that's too wordy. So I've compromised with .only() which is 7 characters compared to [0] which is three.

File: kart/point_cloud/import_.py (hunk 3 above)
Comment (olsen232) [reply kart/point_cloud/import_.py:147]: I could but there's no real upside - it doesn't even make the code simpler

**Review by craigds — APPROVED** [review]
